src/Signature/SignatureV4.php

```diff
@@ -292,7 +292,7 @@ private function getCanonicalizedQuery(array $query)
 
     private function convertToTimestamp($dateValue, $relativeTimeBase = null)
     {
-        if ($dateValue instanceof \DateTime) {
+        if ($dateValue instanceof \DateTimeInterface) {
```

Comment:
Can you add test(s) that pass in objects that aren't `\DateTime` but extend `\DateTimeInterface`? Looks like that's only [`\DateTimeImmutable`](http://php.net/manual/en/class.datetimeimmutable.php).

Reply:
Well, any object that implements that interface from scratch, or extend \DateTimeImmutable would also match. Personally I'm not entirely sure what we should be testing here. \DateTime implements \DateTimeInterface and already passes the tests.

Reply:
The thought is that the current tests already passed with this line as `\DateTime`, the addition of a test with `\DateTimeImmutable` expresses that the change is working as intended.